google-cloud-bigquery/lib/google/cloud/bigquery/project.rb

```diff
@@ -1096,6 +1270,174 @@ def encryption kms_key: nil
+        #   The following values are supported:
+        #
+        #   * `csv` - CSV
+        #   * `json` - [Newline-delimited JSON](http://jsonlines.org/)
+        #   * `avro` - [Avro](http://avro.apache.org/)
+        # @param [String] compression The compression type to use for exported
+        #   files. Possible values include `GZIP` and `NONE`. The default value
+        #   is `NONE`.
+        # @param [String] delimiter Delimiter to use between fields in the
+        #   exported data. Default is <code>,</code>.
+        # @param [Boolean] header Whether to print out a header row in the
+        #   results. Default is `true`.
+        # @param [String] job_id A user-defined ID for the extract job. The ID
+        #   must contain only letters (a-z, A-Z), numbers (0-9), underscores
+        #   (_), or dashes (-). The maximum length is 1,024 characters. If
+        #   `job_id` is provided, then `prefix` will not be used.
+        #
+        #   See [Generating a job
+        #   ID](https://cloud.google.com/bigquery/docs/managing-jobs#generate-jobid).
+        # @param [String] prefix A string, usually human-readable, that will be
+        #   prepended to a generated value to produce a unique job ID. For
+        #   example, the prefix `daily_import_job_` can be given to generate a
+        #   job ID such as `daily_import_job_12vEDtMQ0mbp1Mo5Z7mzAFQJZazh`. The
+        #   prefix must contain only letters (a-z, A-Z), numbers (0-9),
+        #   underscores (_), or dashes (-). The maximum length of the entire ID
+        #   is 1,024 characters. If `job_id` is provided, then `prefix` will not
+        #   be used.
+        # @param [Hash] labels A hash of user-provided labels associated with
+        #   the job. You can use these to organize and group your jobs. Label
+        #   keys and values can be no longer than 63 characters, can only
+        #   contain lowercase letters, numeric characters, underscores and
+        #   dashes. International characters are allowed. Label values are
+        #   optional. Label keys must start with a letter and each label in the
+        #   list must have a different key.
+        # @yield [job] a job configuration object
+        # @yieldparam [Google::Cloud::Bigquery::ExtractJob::Updater] job a job
+        #   configuration object for setting additional options.
+        #
+        # @return [Google::Cloud::Bigquery::ExtractJob]
+        #
+        # @example
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #
+        #   table_id = "bigquery-public-data.samples.shakespeare"
+        #   extract_job = bigquery.extract_job table_id,
+        #                                      "gs://my-bucket/shakespeare.csv"
+        #
+        # @!group Data
+        #
+        def extract_job table, extract_url, format: nil, compression: nil,
+                        delimiter: nil, header: nil, dryrun: nil, job_id: nil,
```

Comment:
Ditto regarding dryrun. I think setting this to true makes the API return an error code.

Reply:
I could not confirm this. I see that setting `dryrun: true` changes the behavior one would expect: The job completes successfully, but the GCS file is *not* created. Can you confirm for us?

Reply:
I also tried running `Table#load_job` with `dryrun: true` and it works as expected: The job completes successfully, but `job.output_rows` is `nil`.

Reply:
However, all this is just empirical observation. The [API docs]() for `configuration.dryRun` state: 

> Behavior of non-query jobs is undefined.

So, I think I should remove/deprecate `dryrun` from all job operations except query. Anyone disagree?

Reply:
I do not disagree. We can always un-deprecate them if they become supported.

Reply:
I agree with deprecation of dryrun for non-query jobs.

> But what is really amazing is that our #query_job methods do not have the dryrun param!

Wow!